### PR TITLE
Swift: Fix missing results in swift/cleartext-storage-database

### PR DESCRIPTION
**Pull request by geoffw0**
Fix missing results in `swift/cleartext-storage-database`.  This is something of a change in philosophy, that any write to a class derived from `RealmSwiftObject` (i.e. something that is to be use with a Realm database) is now considered a sink.  We no longer look for evidence (that we may or may not reliably find) that the `RealmSwiftObject` is truly written to the database, because why would it not be?

### CODE REVIEW

**Review by redsun82 — APPROVED** [review]
LGTM!

File: swift/ql/src/queries/Security/CWE-311/CleartextStorageDatabase.ql [hunk 1]
```diff
@@ -43,22 +43,14 @@ class CoreDataStore extends Stored {
...
-      c.getAMember() = f and
-      f.getName() = "create(_:value:update:)" and
-      call.getStaticTarget() = f and
-      call.getArgument(1).getExpr() = this
+    // any write into a class derived from `RealmSwiftObject` is a sink. For
+    // example in `realmObj.data = sensitive` the post-update node corresponding
+    // with `realmObj.data` is a sink.
+    exists(ClassDecl cd, Expr e |
+      cd.getABaseTypeDecl*().getName() = "RealmSwiftObject" and
+      this.(DataFlow::PostUpdateNode).getPreUpdateNode().asExpr() = e and
```
Comment (MathiasVP) [inline swift/ql/src/queries/Security/CWE-311/CleartextStorageDatabase.ql:51]: A small note: You could replace line 44 with:
```ql
class RealmStore extends Stored instanceof DataFlow::PostUpdateNode
```
and replace `this.(DataFlow::PostUpdateNode)` with `super`.

File: swift/ql/src/queries/Security/CWE-311/CleartextStorageDatabase.ql [hunk 2]
```diff
@@ -20,7 +20,7 @@ import DataFlow::PathGraph
 /**
  * An `Expr` that is stored in a local database.
```
Comment (MathiasVP) [inline swift/ql/src/queries/Security/CWE-311/CleartextStorageDatabase.ql:21]: This comment is now slightly wrong. `Stored` is no longer an `Expr`, but a `DataFlow::Node`.

File: swift/ql/src/queries/Security/CWE-311/CleartextStorageDatabase.ql (hunk 1 above)
Comment (geoffw0) [reply swift/ql/src/queries/Security/CWE-311/CleartextStorageDatabase.ql]: Yep, that's slightly more concise.  Updated.

File: swift/ql/src/queries/Security/CWE-311/CleartextStorageDatabase.ql (hunk 2 above)
Comment (geoffw0) [reply swift/ql/src/queries/Security/CWE-311/CleartextStorageDatabase.ql]: Good point.  Updated.

**Comment by geoffw0** [comment]
Thanks for the reviews, comments addressed.

**Review by MathiasVP — APPROVED** [review]
LGTM!

**Pull request merged**